### ISSUE TITLE
Submitting tasks

### ISSUE THREAD

**Issue by simon-tarr** [Azure/doAzureParallel#300]
Before submitting a bug please check the following:
- [ ] Start a new R session
- [ ] Check your credentials file
- [ ] Install the latest doAzureParallel package
- [ ] Submit a minimal, reproducible example 
- [ ] run `sessionInfo()`

**Description**

**Instruction to repro the problem if applicable**

**Comment by brnleehng** [comment]
Hi Simon,

For submitting a large number of tasks, I'll look into increasing the throughput of task submission similar to the other Azure Batch SDKs.

https://docs.microsoft.com/en-us/azure/batch/large-number-tasks#increase-throughput-of-task-submission

Thanks,
Brian

**Comment by simon-tarr** [comment]
Hi @brnleehng, thanks for the reply. It sounds like it could be a great enhancement to the package.

**Comment by simon-tarr** [comment]
Hi Brian,

I was just wondering how complex this is to implement? Is it possible to get a rough ETA on when this could be expected to go live, assuming it's possible to increase the throughput in the first place? In addition, is this a back-end update that will just start working after submitting a job, rather than needing a package update within doAzureParallel?

Many thanks,
Simon

**Comment by darylmsft** [comment]
We should be able to get the task factory to call AddTaskCollection and handle its complexity by the end of Oct.  That will get about 100x improvement.

**Comment by simon-tarr** [comment]
> We should be able to get the task factory to call AddTaskCollection and handle its complexity by the end of Oct. That will get about 100x improvement.

Wow, 100x is a serious improvement.

I guess this leaves me to figure out whether waiting until the end of October and then running all my tasks will be quicker than letting everything chug along slowly as things currently stand. Currently about 75% of the time each iteration takes to run can be explained by the time taken to submit tasks but if this gets improved 100x, there will essentially be no waiting. Decisions...

**Comment by simon-tarr** [comment]
Yeah it’s a bit mad that this throughput wasn’t included from day 1, really. I’m spending thousands of dollars more than I should be because, cumulatively, submitting tasks is taking hundreds and hundreds of hours.

**Comment by simon-tarr** [comment]
Hello, any news on the status of this feature? Thanks!

**Comment by Pullarg** [comment]
would be excellent to know, running 800k jobs 12 hours and still submitting, at this point spinning up a 64core machine and using doparallel

**Comment by simon-tarr** [comment]
> would be excellent to know, running 800k jobs 12 hours and still submitting, at this point spinning up a 64core machine and using doparallel would be better

I reckon half of the $20,000 I've spent over the last 6 weeks has been on waiting for tasks to submit...
